whirlpool/appliancesmanager.py

```diff
@@ -138,10 +135,14 @@ async def _get_shared_appliances(self) -> bool:
 
             return True
 
-    async def fetch_appliances(self):
+    async def fetch_appliances(self) -> bool:
         account_id = await self._auth.get_account_id()
         if not account_id:
             return False
+
+        # Invalidate cached property
+        self.__dict__.pop("all_appliances", None)
```

Comment:
we should move this back to `_add_appliance`, since that is the method that actually adds a new appliance. having it there prevents us from introducing a bug in the future in case it gets called from more methods.

the performance impact should be negligible anyway.